src/compiler/visitorPublic.ts

```diff
@@ -793,6 +793,12 @@ namespace ts {
                     nodeVisitor(node.operatorToken, tokenVisitor, isBinaryOperatorToken),
                     nodeVisitor(node.right, visitor, isExpression));
 
+            case SyntaxKind.PrivateIdentifierInInExpression:
+                return factory.updatePrivateIdentifierInInExpression(<PrivateIdentifierInInExpression>node,
+                    nodeVisitor((<PrivateIdentifierInInExpression>node).name, visitor, isExpression),

```

Comment:
Should this use `isPrivateIdentifier` check?

```suggestion
                    nodeVisitor((<PrivateIdentifierInInExpression>node).name, visitor, isPrivateIdentifier),

```

Reply:
If my memory serves correctly: no. As I believe this check is also done on the transformed AST where `PrivateIdentifierInInExpression` becomes a methodCall.

This was over a month ago so I may be getting this mixed up with a different bit of code. I'll try it out again and confirm either way.

Reply:
Played around to jog my memory. The reason this was `isExpression` and not `isPrivateIdentifier` was because when the identifier is invalid (e.g. used outside of a class) the classFields transformer emits an empty plain Identifier.

I've changed this to `isMemberName` now to match both private and standard identifiers.